app.js

```diff
@@ -199,8 +230,35 @@ const play = async () => {
         player.x = currentPosition.x;
         player.y = currentPosition.y;
       }
+      if (food.x === wall.x && food.y === wall.y) {
+        food.x = Math.floor(Math.random() * boardWidth);
+        food.y = Math.floor(Math.random() * boardHeight);
+      }
     });
+    if (food.x === player.x && food.y === player.y) {
+      verifyFoodPosition(food.x, food.y);
+      player.hunger >= 70 ? (player.hunger = 100) : (player.hunger += 30);
+    }
   });
+
+  const verifyFoodPosition = (x, y) => {
```

Comment:
We do not need to verify the position if we draw from valid-only positions. We need to draw a tile from floor tiles and get the x, y of it :)

Reply:
I made a commit with solution for this problem :)